XenCert/StorageHandler.py

```diff
@@ -3170,3 +3169,105 @@ def ControlPathStressTests(self):
         XenCertPrint("Checkpoints: %d, totalCheckPoints: %s" % (checkPoint, totalCheckPoints))
 
         return (retVal, checkPoint, totalCheckPoints)
+
+    def DataIntegrityTests(self):
+        retVal = None
+        checkPoint = 0
+        totalCheckPoints = 11
+
+        vm_uuid = StorageHandlerUtil._get_localhost_uuid()
+        XenCertPrint("Got vm_uuid as %s" % vm_uuid)
+        vm_ref = self.session.xenapi.VM.get_by_uuid(vm_uuid)
+        sr_ref = None
+        vdi_ref = None
+        vbd_ref = None
+
+        try:
+            #1) Create SR
+            (retVal, sr_ref, dconf) = self.Create()
+            Print("Created SR")
+            checkPoint += 1
+
+            #2) Create 4GB VDI in SR
+            # XXX: Make Create_VDI() return vdi_ref on success and raise an
+            #      exception on failure. This is ugly and stoopid.
+            (retVal, vdi_ref) = self.Create_VDI(
+                    sr_ref,
+                    4 * StorageHandlerUtil.GiB)
+            if not retVal:
+                raise Exception('Error in VDI creation: %s' % vdi_ref)
```

Comment:
Let's keep this as-is for now and address it as part of a more comprehensive cleanup.